### DISCUSSION TITLE
Use Universal Provider with TRON

### DISCUSSION THREAD

**Discussion by kandrukKV** [WalletConnect/walletconnect-monorepo#4434] [Developer Support]
Hi, friends!

I'm trying to use @walletconnect/universal-provider to connect to the network TRON. 

I managed to connect via web3Modal and Trust.

`async connect() {

    if (!this.universalProvider) {

      this.universalProvider = await UniversalProvider.init({
        logger: 'info',
        relayUrl: 'wss://relay.walletconnect.com',
        projectId: this.projectId
      })

      this.universalProvider.on("display_uri", (uri:string) => {
        this.web3Modal.openModal({ uri });
      });

      if (!this.universalProvider.session) {
        await this.universalProvider.connect({
          namespaces: {
            tron: {
              chains: ['tron:0x2b6653dc'],
              methods: ['tron_signTransaction', 'tron_signMessage'],
              events: [],
              rpcMap: {
                '0x2b6653dc': 'https://api.trongrid.io'
              }
            }
          },
          skipPairing: false
        })
      }

      this.web3Modal.closeModal();
    }`

Now I want to get at least some information from

`const web3Provider = new ethers.providers.Web3Provider(this.universalProvider)`

`console.log('web3Provider', await web3Provider.listAccounts())`

But I keep seeing a mistake: TypeError: Cannot read properties of undefined (reading 'getDefaultChain')

![1](https://github.com/WalletConnect/walletconnect-monorepo/assets/54434777/be1bcdce-535f-41d0-8a7b-22087485cfa4)

Please tell me what can be done to make it work?

**Comment by mouhand999** [comment]
Help
